platform-sdk/swirlds-state-api/src/main/java/com/swirlds/state/spi/WritableSingletonStateBase.java

```diff
@@ -73,26 +60,23 @@ public void registerListener(@NonNull final SingletonChangeListener<T> listener)
 
     @Override
     public T get() {
-        // Possible pattern: "put" and then "get". In this case, "read" should be false!! Otherwise,
-        // we invalidate tx when we don't need to
-        final var currentValue = currentValue();
-        if (currentValue != null) {
-            // C.f. https://github.com/hashgraph/hedera-services/issues/14582; in principle we should
-            // also return null here if value is NULL_VALUE, but in production with the SingletonNode
-            // backing store, null values are never actually set so this doesn't matter
-            return currentValue;
+        // If there is a modification, then we've already done a "put" or "remove"
+        // and should return based on the modification
+        if (isModified()) {
+            return modification;
+        } else {
+            return super.get();
         }
-        return super.get();
     }
 
     @Override
     public void put(T value) {
-        this.value = value == null ? NULL_VALUE : value;
+        this.modification = value;
```

Comment:
Null values need to be handled separately. If a value was not null, then `put(null)` was called, `modification` is null, and `isModified()` below returns false, which is incorrect

Reply:
Good catch 🙌🏻 Addressed, please look into it again 🙂